src/Grid/grid.jl

```diff
@@ -23,96 +23,111 @@ immutable Node{dim, T}
     x::Vec{dim, T}
 end
 Node{dim, T}(x::NTuple{dim, T}) = Node(Vec{dim, T}(x))
+getcoordinates(n::Node) = n.x
+
+#=
+#A bit redundant but perhaps we want to put more in here:
+immutable Face
+    onboundary::Bool
+end
+onboundary(f::Face) = f.onboundary
+Face() = Face(false)
+=#
 
 """
 A `Cell` is a sub-domain defined by a collection of `Node`s as it's vertices.
 """
-immutable Cell{dim, N}
+immutable Cell{dim, N, M}
     nodes::NTuple{N, Int}
+    onboundary::NTuple{M, Bool}
```

Comment:
We could say what nodes are on the boundary and then `onboundary(cell, node)` checks if all the nodes are on the boundary.